vehicle/src/Vehicle/Backend/Isabelle/Compile.hs

```diff
@@ -0,0 +1,897 @@
+module Vehicle.Backend.Isabelle.Compile
+  ( IsabelleOptions (..),
+    compileProgToIsabelle,
+  )
+where
+
+import Data.Maybe (mapMaybe)
+import Data.Foldable (fold)
+import Data.List.NonEmpty (NonEmpty ((:|)))
+import Data.List.NonEmpty qualified as NonEmpty
+import Data.Set (Set)
+import Data.Set qualified as Set
+import Data.Text (Text)
+import Data.Text qualified as Text
+import Data.Text.Internal.Read qualified as Text.Read
+import GHC.Real (denominator, numerator)
+import Prettyprinter hiding (hcat, hsep, vcat, vsep)
+import Prettyprinter.Render.Text (renderStrict)
+import System.FilePath (takeBaseName)
+import Vehicle.Compile.Error
+import Vehicle.Compile.Prelude hiding (Module)
+import Vehicle.Compile.Print
+import Vehicle.Data.Builtin.Decidability
+import Vehicle.Data.Builtin.Interface (Accessor (..))
+import Vehicle.Data.Builtin.Standard hiding (TensorType)
+import Vehicle.Data.Code.Expr ()
+import Vehicle.Data.Code.Interface (IsArgs (..), VecLitArgs (..))
+import Vehicle.Data.Universe (UniverseLevel (..))
+import Vehicle.Data.Variable.Bound.Context.Name (MonadNameContext, addNameToContext, ixToProperName, runFreshNameBoundContextT, getNameContext)
+import Vehicle.Syntax.Builtin
+import Vehicle.Syntax.Sugar
+  ( BinderType (..),
+    LetBinder,
+    foldBinders,
+    foldDeclBinders,
+  )
+import Vehicle.Syntax.Tensor
+  ( Tensor (..),
+    TensorShape,
+    foldMapTensor,
+    shapeOf,
+    toList,
+  )
+
+--------------------------------------------------------------------------------
+-- Isabelle-specific options
+
+data IsabelleOptions = IsabelleOptions
+  { output :: Maybe FilePath,
+    localeName :: Maybe String
+  }
+
+currentPhase :: Doc ()
+currentPhase = "compilation to Isabelle"
+compileProgToIsabelle :: (MonadCompile m) => Prog DecidabilityBuiltin -> IsabelleOptions -> m (Doc a)
```

Comment:
Is there a reason the formatter isn't automatically putting an empty line between `currentPhase` and `compileProgToIsabelle`?